nlmod/observations.py

```diff
@@ -0,0 +1,232 @@
+    bary = np.einsum("njk,nk->nj", temp[:, :d, :], delta)
+    return vertices, np.hstack((bary, 1 - bary.sum(axis=1, keepdims=True)))
+
+
+def interpolate_vertices_weights(values, vertices, weights, fill_value=np.nan):
+    """Interpolate values at locations given computed vertices and weights.
+
+    As calculated by compute_barycentric_interpolation_weights function [2]_.
+
+    Parameters
+    ----------
+    values : np.array
+        array containing values to interpolate
+    vtx : np.array
+        array containing interpolation vertices, see
+        compute_barycentric_interpolation_weights()
+    wts : np.array
+        array containing interpolation weights, see
+        compute_barycentric_interpolation_weights()
+    fill_value : float
+        fill value for points that have to be extrapolated (e.g. at or
+        beyond edges of the known points)
+
+    Returns
+    -------
+    arr: np.array
+        array containing interpolated values at locations as given by
+        vtx and wts
+
+    References
+    ----------
+    .. [2] https://stackoverflow.com/questions/20915502/
+    speedup-scipy-griddata-for-multiple-interpolations-between-two-irregular-grids
+    """
+    ret = np.einsum("nj,nj->n", np.take(values, vertices), weights)
+    ret[np.any(weights < 0, axis=1)] = fill_value
+    return ret
+
+
+def interpolate_points(da, xi, yi, x="x", y="y"):
+    """Interpolate values at locations given by x and y coordinates.
+
+    This function only works for 2D (planar) data, both for structured
+    and vertex grid types.
+
+    Parameters
+    ----------
+    da : np.array or xr.DataArray
+        array containing values to interpolate
+    xi : np.array
+        x-coordinates of the points to interpolate
+    yi : np.array
+        y-coordinates of the points to interpolate
+    x : str or np.array, optional
+        x-coordinates of the array, default is "x", which assumes da is
+        an xr.DataArray with an x-coordinate named "x"
+    y : str or np.array, optional
+        y-coordinates of the array, default is "y", which assumes da is
+        an xr.DataArray with an y-coordinate named "y"
+
+    Returns
+    -------
+    np.array
+        array containing interpolated values at locations defined by x and y
+    """
+    # interpolation points
+    xyi = np.vstack([xi, yi]).T
+    # data coordinates
+    if isinstance(x, str):
+        x = da[x].values
+    if isinstance(y, str):
+        y = da[y].values
+
+    if grid.is_structured(da):
+        assert da.ndim == 2, "DataArray must be 2D!"
+        xy = np.vstack([arr.ravel() for arr in np.meshgrid(x, y)]).T
+    elif grid.is_vertex(da):
+        assert da.ndim == 1, "DataArray must be 1D!"
+        xy = np.vstack([x, y]).T
+    else:
+        raise ValueError("da must be structured or vertex grid")
+    vertices, weights = compute_interpolation_weights(xy, xyi, d=2)
+    return interpolate_vertices_weights(da.values, vertices, weights)
+
+
+def interpolate_points_ds(
```

Comment:
Maybe a more descriptive name is `interpolate_to_points`

Reply:
or `interpolate_da_to_points`

Reply:
Yea, good suggestion, those are much better. I went for interpolate to points.